paasta_tools/autoscaling_lib.py

```diff
@@ -775,19 +778,18 @@ class ClusterAutoscalingError(Exception):
     pass
 
 
-def get_instance_id_from_ip(ip):
+def get_instance_id_from_ip(ip, instance_descriptions):
```

Comment:
So definitely agree this should just be `get_instance_from_ip` and return the whole instance.

Not so sure on moving the check for the number of instances returned. Don't think this function should ever return more than one instance. It might return `None` if it can't find the IP you specify. I guess it would be sensible to check that we never get more than one instance (and maybe raise an error) but I'd still say that check should live in this function.